src/js/sw.js

```diff
@@ -3,18 +3,18 @@ import { BackgroundSyncPlugin } from 'workbox-background-sync';
 import { registerRoute, Route } from 'workbox-routing';
 import { NetworkOnly, NetworkFirst, CacheFirst } from 'workbox-strategies';
 import { CacheableResponsePlugin } from 'workbox-cacheable-response';
-import { PrecacheFallbackPlugin, precacheAndRoute } from 'workbox-precaching';
+import { PrecacheFallbackPlugin, precacheAndRoute, cleanupOutdatedCaches } from 'workbox-precaching';
 
 // Ensure that updates to the underlying service worker take effect immediately
 // for both the current client and all other active clients.
 self.skipWaiting();
 clientsClaim();
 
-// precache webpack manifest
-precacheAndRoute(self.__WB_MANIFEST);
+if (self.location.hostname !== 'localhost') {
```

Comment:
almost of the changes in this file are due to this wrapping `if`
Essentially if we let this worker run for localhost it will heavily cache the esbuild dev version of vite and making development nearly impossible.  Testing SW will be a pain, but not impossible.